src/gkmodel/composite_model.py

```diff
@@ -2,59 +2,101 @@
     composite_model
     ~~~~~~~~~~~~~~~
 """
-from typing import List, Union, Dict, Tuple
-import pandas as pd
+from pathlib import Path
+from typing import List, Dict
 import numpy as np
 from copy import deepcopy
 
 from mrtool import MRData, LinearCovModel
-from .model import OverallModel, StudyModel
+from .model import NodeModel, OverallModel, StudyModel
 
 
 class StagewiseModel:
 
+    def __init__(self,
+                 data: MRData,
+                 node_models: List[NodeModel]):
+        self.node_models = node_models
+        self.num_models = len(node_models)
+        self.data_list = [data]
+
+    def _get_next_data(self, model: NodeModel):
+        pred = model.predict(self.data_list[-1])
+        resi = self.data_list[-1].obs - pred
+        data = deepcopy(self.data_list[-1])
+        data.obs = resi
+
+        self.data_list.append(data)
+
+    def fit_model(self):
+        for model in self.node_models:
+            model.attach_data(self.data_list[-1])
+            model.fit_model()
+            self._get_next_data(model)
```

Comment:
I think we may not want to do this for the last model, since that would make the `len(data_list)` longer than the models. (or if we do not care about that, that's okay too.)

Reply:
Thanks very much! added the changes :)